src/index.ts

```diff
@@ -265,6 +265,7 @@ const SplunkRum: SplunkOtelWebType = {
     const instrumentations = INSTRUMENTATIONS.map(({ Instrument, confKey, disable }) => {
       const pluginConf = getPluginConfig(processedOptions.instrumentations[confKey], pluginDefaults, disable);
       if (pluginConf) {
+        // @ts-expect-error Can't mark in any way that processedOptions.instrumentations[confKey] is of specifc config type
```

Comment:
Can you elaborate on this? I think it would be good to have typings work well in config.

Reply:
Basically typescript detected the type of `processedOptions.instrumentations[confKey]` to be `boolean | XMLHttpRequestInstrumentationConfig | InstrumentationConfig | SplunkUserInteractionInstrumentationConfig | ...` while Instrument constructor is now expecting an object that covers all of those different configs (`XMLHttpRequestInstrumentationConfig & InstrumentationConfig & SplunkUserInteractionInstrumentationConfig & ...`)

leading to error:

```
Argument of type 'InstrumentationConfig | FetchInstrumentationConfig | SplunkUserInteractionInstrumentationConfig | SplunkPostDocLoadResourceInstrumentationConfig | XMLHttpRequestInstrumentationConfig' is not assignable to parameter of type 'InstrumentationConfig & XMLHttpRequestInstrumentationConfig & SplunkUserInteractionInstrumentationConfig & SplunkPostDocLoadResourceInstrumentationConfig & FetchInstrumentationConfig & SplunkWebSocketInstrumentationConfig'.
  Type 'FetchInstrumentationConfig' is not assignable to type 'InstrumentationConfig & XMLHttpRequestInstrumentationConfig & SplunkUserInteractionInstrumentationConfig & SplunkPostDocLoadResourceInstrumentationConfig & FetchInstrumentationConfig & SplunkWebSocketInstrumentationConfig'.
    Type 'FetchInstrumentationConfig' is not assignable to type 'XMLHttpRequestInstrumentationConfig'.
      Types of property 'applyCustomAttributesOnSpan' are incompatible.
        Type 'FetchCustomAttributeFunction' is not assignable to type 'XHRCustomAttributeFunction'.
```

Fix would probably be to instead of mapping an array making a function and having it call another function that would do this one instrumentation at a time.